### ISSUE TITLE
Html/Xml file does not parse unless we use -force-language flag with xml language

### ISSUE THREAD

**Issue by kevingnet** [pmd/pmd#3521]
Html/Xml file does not parse unless we use --force-language flag with xml language

**Affects PMD Version:** 6.36-38

**Description:**

**Exception Stacktrace:**

```
# Copy-paste the stack trace here
net.sourceforge.pmd.PMDException: Error while parsing /usr/local/google/home/guerrak/code/git/shg-eng-excellence/apexrules/src/test/com/google/rules/pmd/html/ComponentsLightningMustBeDocumented.html
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCodeWithoutCache(SourceCodeProcessor.java:124)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:100)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:62)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:85)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:29)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: net.sourceforge.pmd.lang.java.ast.ParseException: Encountered " "<" "< "" at line 1, column 1.
Was expecting one of:
    <EOF> 
    "import" ...
    "\u001a" ...
    "~[]" ...
    
	at net.sourceforge.pmd.lang.java.ast.JavaParser.generateParseException(JavaParser.java:13532)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.jj_consume_token(JavaParser.java:13416)
	at net.sourceforge.pmd.lang.java.ast.JavaParser.CompilationUnit(JavaParser.java:557)
	at net.sourceforge.pmd.lang.java.AbstractJavaParser.parse(AbstractJavaParser.java:62)
	at net.sourceforge.pmd.lang.AbstractParser.doParse(AbstractParser.java:44)
	at net.sourceforge.pmd.SourceCodeProcessor.parse(SourceCodeProcessor.java:136)
	at net.sourceforge.pmd.SourceCodeProcessor.processSource(SourceCodeProcessor.java:200)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCodeWithoutCache(SourceCodeProcessor.java:118)
	... 10 more


**Code Sample demonstrating the issue:**
Load an xml file with any basic content... <html> </html>
```

```

**Steps to reproduce:**
run PMD's run.sh with a file that derives from AbstractXmlRule

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by adangel** [comment]
According to this list - which should be up to date - only XML is supported: https://pmd.github.io/latest/pmd_userdocs_cli_reference.html#supported-languages

I assume, you have used the file extension ".html" for your html file? Then, as no language module supports ".html" files, we fall back trying to parse it as a Java file - that's the exception stacktrace you are seeing.

We might change this behaviour in the future to display a nicer message, such as "HTML is not supported".

So, it's understandable, that you need to use `-force-language xml` in order to analyze html files (which use the extension html).
But it might only work, if you html file is using strict xhtml (html5 is not xml)

you would also need rules, of course, as otherwise analyzing a html file without a rule is useless

On the other side: XML is supported and should work without `-force-language` as long as you use "xml" as a file extension. If your xml file uses a different file extension, that you need to use `-force-language`.

**Comment by oowekyala** [comment]
HTML support has been implemented as a response to #3941 and was shipped in PMD 6.45.0.
